code/game/machinery/syndicatebomb.dm

```diff
@@ -292,6 +342,128 @@
 	explosion(src.loc, 1, 2, 4, flame_range = 2) //Identical to a minibomb
 	qdel(src)
 
+#define MAX_BEAKERS 4
+#define SPREAD_RANGE 7
+
+/obj/item/weapon/bombcore/chemical
+	name = "chemical payload"
+	desc = "An explosive payload designed to spread chemicals, dangerous or otherwise, across a large area. It is able to hold up to four chemical containers, and must be loaded before use."
+	origin_tech = "combat=4;materials=3"
+	icon_state = "chemcore"
+	var/list/beakers = list()
+
+/obj/item/weapon/bombcore/chemical/New()
+	..()
+	create_reagents(300*MAX_BEAKERS)
+
+/obj/item/weapon/bombcore/chemical/detonate()
+	if(adminlog)
+		message_admins(adminlog)
+		log_game(adminlog)
+
+	if(!reagents)
+		return
+
+	var/has_reagents
+	for(var/obj/item/I in beakers)
+		if(I.reagents.total_volume)
+			has_reagents = 1
+
+	if(!has_reagents)
+		playsound(loc, 'sound/items/Screwdriver2.ogg', 50, 1)
+		return
+
+	playsound(loc, 'sound/effects/bamf.ogg', 75, 1)
+
+	mix_reagents()
+
+	if(reagents.total_volume)	//The possible reactions didnt use up all reagents, so we spread it around.
+		var/datum/effect_system/steam_spread/steam = new /datum/effect_system/steam_spread()
+		steam.set_up(10, 0, get_turf(src))
+		steam.attach(src)
+		steam.start()
+
+		var/list/viewable = view(SPREAD_RANGE, get_turf(src))
+		var/list/accessible = can_flood_from(get_turf(src), SPREAD_RANGE)
+		var/list/reactable = accessible
+		var/mycontents = GetAllContents()
+		for(var/turf/T in accessible)
+			for(var/atom/A in T.GetAllContents())
+				if(A in mycontents) continue
+				if(!(A in viewable)) continue
+				reactable |= A
+		if(!reactable.len) //Nothing to react with. Probably means we're in nullspace.
+			qdel(src)
+			return
+		var/fraction = 1/reactable.len
+		for(var/atom/A in reactable)
+			reagents.reaction(A, TOUCH, fraction)
```

Comment:
the alternative is hacky, or to implement global chem_mixing procs and change grenades to use those instead. This is faster and more efficient than either.